[Bug] [connector] [dm] [elasticsearch7] 找不到依赖包

### Search before asking

- [X] I had searched in the [issues](https://github.com/DTStack/chunjun/issues) and found no similar issues.


### What happened

找不到依赖包


```

2023-06-05 10:50:10,329 INFO  org.apache.flink.runtime.executiongraph.ExecutionGraph       [] - Source: source[1] -> Sink: sink[2] (1/1) (fdd3a2040c136b635aa9b7bfdab2dc68_cbc357ccb763df2852fee8c4fc7d55f2_0_0) switched from INITIALIZING to FAILED on container_e225_1683781588671_701059_01_000002 @ 10.x.x.x (dataPort=33044).
java.lang.RuntimeException: java.lang.ClassNotFoundException: dm.jdbc.driver.DmDriver
	at com.dtstack.chunjun.util.ClassUtil.forName(ClassUtil.java:34) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.connector.jdbc.util.JdbcUtil.getConnection(JdbcUtil.java:97) ~[chunjun-connector-hive-master.jar:?]
	at com.dtstack.chunjun.connector.jdbc.sink.JdbcOutputFormat.getConnection(JdbcOutputFormat.java:527) ~[chunjun-connector-hive-master.jar:?]
	at com.dtstack.chunjun.connector.jdbc.sink.JdbcOutputFormat.openInternal(JdbcOutputFormat.java:91) ~[chunjun-connector-hive-master.jar:?]
	at com.dtstack.chunjun.sink.format.BaseRichOutputFormat.open(BaseRichOutputFormat.java:280) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.sink.DtOutputFormatSinkFunction.open(DtOutputFormatSinkFunction.java:90) ~[chunjun-core.jar:?]
	at org.apache.flink.api.common.functions.util.FunctionUtils.openFunction(FunctionUtils.java:34) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.api.operators.AbstractUdfStreamOperator.open(AbstractUdfStreamOperator.java:100) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.table.runtime.operators.sink.SinkOperator.open(SinkOperator.java:58) ~[flink-table-runtime-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.initializeStateAndOpenOperators(RegularOperatorChain.java:107) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restoreGates(StreamTask.java:726) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$SynchronizedStreamTaskActionExecutor.call(StreamTaskActionExecutor.java:100) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restoreInternal(StreamTask.java:702) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restore(StreamTask.java:669) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:935) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:904) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:728) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:550) ~[flink-dist-1.16.1.jar:1.16.1]
	at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_60]
Caused by: java.lang.ClassNotFoundException: dm.jdbc.driver.DmDriver
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[?:1.8.0_60]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_60]
	at org.apache.flink.util.FlinkUserCodeClassLoader.loadClassWithoutExceptionHandling(FlinkUserCodeClassLoader.java:67) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.util.FlinkUserCodeClassLoader.loadClass(FlinkUserCodeClassLoader.java:51) ~[flink-dist-1.16.1.jar:1.16.1]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_60]
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_60]
	at java.lang.Class.forName(Class.java:348) ~[?:1.8.0_60]
	at com.dtstack.chunjun.util.ClassUtil.forName(ClassUtil.java:31) ~[chunjun-core.jar:?]
	... 18 more
2023-06-05 10:50:10,357 INFO  org.apache.flink.runtime.resourcemanager.slotmanager.DeclarativeSlotManager [] - Clearing resource requirements of job 78be9e309a665dcde9e747268f09a81b

```



```
Caused by: org.apache.flink.table.api.ValidationException: Cannot discover a connector using option: 'connector'='elasticsearch7-x'
	at org.apache.flink.table.factories.FactoryUtil.enrichNoMatchingConnectorError(FactoryUtil.java:735)
	at org.apache.flink.table.factories.FactoryUtil.discoverTableFactory(FactoryUtil.java:709)
	at org.apache.flink.table.factories.FactoryUtil.createDynamicTableSink(FactoryUtil.java:264)
	... 35 more
Caused by: org.apache.flink.table.api.ValidationException: Could not find any factory for identifier 'elasticsearch7-x' that implements 'org.apache.flink.table.factories.DynamicTableFactory' in the classpath.
```

### What you expected to happen

--

### How to reproduce

--

### Anything else

_No response_

### Version

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/chunjun-connectors/chunjun-connector-dm/pom.xml
+++ b/chunjun-connectors/chunjun-connector-dm/pom.xml
@@ -55,6 +55,10 @@
 		<plugins>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-antrun-plugin</artifactId>
 			</plugin>
 		</plugins>

--- a/chunjun-connectors/chunjun-connector-elasticsearch7/pom.xml
+++ b/chunjun-connectors/chunjun-connector-elasticsearch7/pom.xml
@@ -71,6 +71,52 @@
 		<plugins>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<version>3.2.0</version>
+				<executions>
+					<execution>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<createDependencyReducedPom>false</createDependencyReducedPom>
+							<relocations>
+								<relocation>
+									<pattern>org.apache.http</pattern>
+									<shadedPattern>com.dtstack.chunjun.connector.elasticsearch7.org.apache.http</shadedPattern>
+								</relocation>
+								<relocation>
+									<pattern>org.elasticsearch</pattern>
+									<shadedPattern>com.dtstack.chunjun.connector.elasticsearch7.org.elasticsearch</shadedPattern>
+								</relocation>
+							</relocations>
+							<artifactSet>
+								<excludes>
+									<exclude>com.data-artisans:*</exclude>
+									<exclude>org.scala-lang:*</exclude>
+									<exclude>io.netty:*</exclude>
+									<exclude>org.slf4j:slf4j-api</exclude>
+									<exclude>log4j:log4j</exclude>
+									<exclude>ch.qos.logback:*</exclude>
+								</excludes>
+							</artifactSet>
+							<filters>
+								<filter>
+									<artifact>*:*</artifact>
+									<excludes>
+										<exclude>META-INF/*.SF</exclude>
+										<exclude>META-INF/*.DSA</exclude>
+										<exclude>META-INF/*.RSA</exclude>
+									</excludes>
+								</filter>
+							</filters>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-antrun-plugin</artifactId>
 			</plugin>
 		</plugins>

